scheduler/test/cook/test/kubernetes/controller.clj

```diff
@@ -72,6 +72,46 @@
       (is (nil? (do-process :missing :pod/unknown)))
       (is (= :missing (do-process :missing :pod/waiting))))))
 
+(deftest test-launch-kill-process
+  (let [pod-name "TestPodName-LKP"
+        cook-expected-state-map (atom {})
+        k8s-actual-state-map (atom {})
+        mock-cc {:api-client nil
+                 :cook-expected-state-map cook-expected-state-map
+                 :k8s-actual-state-map k8s-actual-state-map}
+        extract-cook-expected-state (fn []
+                                      (:cook-expected-state (get @cook-expected-state-map pod-name {})))
+        count-kill-pod (atom 0)]
+    (with-redefs [controller/kill-pod  (fn [_ cook-expected-state-dict _] (swap! count-kill-pod inc) cook-expected-state-dict)
+                  controller/launch-pod (fn [_ cook-expected-state-dict] cook-expected-state-dict)
+                  controller/log-weird-state (fn [_ _ _ _] :illegal_return_value_should_be_unused)
```

Comment:
Why do we need to `with-redefs` `log-weird-state`?

Reply:
Nuked.